google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -186,9 +187,17 @@ public void apply() throws ConfigurationException {
     }
 
     try {
-      if (isRuntimeCustom() && !isValidConfigurationFile(dockerfile.getText())) {
-        throw new ConfigurationException(
-            GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+      if (isRuntimeCustom()) {
+        String dockerfileDir = dockerDirectory.getText();
+        if (dockerfileDir.isEmpty() || !Files.isDirectory(Paths.get(dockerfileDir))) {
+          throw new ConfigurationException(
+              GctBundle.getString("appengine.deployment.error.staging.docker.directory"));
```

Comment:
@etanshaul 
@nbashirbello 
I think this the kind of thing Nikolay suggested we not do in apply methods #1408